waku/mailserver.md

```diff
@@ -62,7 +62,7 @@ The `Cursor` field SHOULD be filled in if a number of envelopes between `Lower`
 
 The envelope MUST be encrypted with a symmetric key agreed between the requester and Mailserver.
 
-If `topic-interest` is used the `Cursor` field MUST be specified for the argument order to be unambiguous. However, it MAY be set to null. `topic-interest` is used to specify which topics a node is interested in. If this is specified, a mailserver MUST NOT send messages that aren't in in that topic. When `topic-interest` is set (even if it an empty array), the `bloom` filter parameter should be ignored.
+If `topics` is used the `Cursor` field MUST be specified for the argument order to be unambiguous. However, it MAY be set to null. `topics` is used to specify which topics a node is interested in. If this is specified, a mailserver MUST NOT send messages that aren't in in that topic. When `topic` is set (even if it an empty array), the `bloom` filter parameter should be ignored.
```

Comment:
> When `topic` is set (even if it an empty array), the `bloom` filter parameter should be ignored.

I am not sure it's possible with RLP. I need to dig into this but at least in Go's RLP implementation, when an array being null is serialized, it is deserialized as an empty array, not a null value. If this is true, I will change this fragment as well.

Reply:
I just confirmed that serializing `nil` produces not-nil* but empty (there is a difference) value so we need to change this fragment to avoid ambiguity.

Additionally:
```go
type Request struct {
    Bloom []byte
}

rNil := Request{}
rNotNil := Request{Bloom: []byte{}}
```

`rNil` and `rNotNil` produces the same output after serialization.

* - actually, there is a way to do it. RLP in Go supports `rlp:"nil"` tag on a field but a field needs to be a pointer. In case of slices, it's a bit more tricky. The slice itself is passed by value but it's only its header which is a pointer + 2 ints. So to use `rlp:"nil"`, a byte slice needs to be defined as `*[]byte`. Still, it's not distinguishable whether the serialized value was nil or empty slice.

I would also support changing this spec to get rid of optional fields at all.